### PR TITLE
feat(react-theme): Add bold font weight

### PR DESCRIPTION
**Pull request by miroslavstastny**
## Current Behavior

- there was no `fontWeightBold` token in react-theme
- there was no `Body2` in Typography
- all Stronger Typography variants used `fontWeightMedium`

## New Behavior

Add `fontWeightBold` token to react theme
image

Add `bold` as a valid option of `weight` prop in `Text`.

Update typography to use `fontWeightBold` in all Stronger variants
image
image
image

Add `Body2` to Typography and also as a component to `react-text`
image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                  |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                  Change |
| :------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `32.876 kB`<br />`10.773 kB` | `32.895 kB`<br />`10.778 kB` |   `19 B` <br />`5 B`  |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                                             |  `11.572 kB`<br />`4.537 kB` |  `11.682 kB`<br />`4.561 kB` | `110 B` <br />`24 B`  |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                                           |  `14.882 kB`<br />`4.977 kB` |  `14.992 kB`<br />`4.995 kB` | `110 B` <br />`18 B`  |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsAllThemes.fixture.js'>Teams: all themes</abbr>                                       |  `29.205 kB`<br />`6.249 kB` |  `29.224 kB`<br />`6.255 kB` |   `19 B` <br />`6 B`  |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsLightTheme.fixture.js'>Teams: Light theme</abbr>                                     |  `17.069 kB`<br />`4.885 kB` |   `17.088 kB`<br />`4.89 kB` |   `19 B` <br />`5 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |  `79.574 kB`<br />`24.123 kB` |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `82.874 kB`<br />`20.608 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `48.283 kB`<br />`13.644 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |   `15.072 kB`<br />`6.016 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `68.464 kB`<br />`19.067 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |   `22.503 kB`<br />`7.153 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |   `23.406 kB`<br />`7.449 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                                                                       |   `23.947 kB`<br />`7.022 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |   `36.396 kB`<br />`9.575 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |  `43.469 kB`<br />`10.812 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |  `39.014 kB`<br />`10.456 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                                                          |   `46.544 kB`<br />`11.84 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |   `51.91 kB`<br />`11.003 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>                                                                                |   `67.47 kB`<br />`19.273 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>                                                                                          |  `63.152 kB`<br />`18.185 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>                                                                              |    `8.461 kB`<br />`3.555 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>                                                                              |    `9.504 kB`<br />`3.896 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr>                                                                            |     `8.562 kB`<br />`3.61 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `72.723 kB`<br />`23.746 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `70.792 kB`<br />`23.233 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.042 kB`<br />`51.964 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |   `84.793 kB`<br />`25.27 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                                                                 |   `16.359 kB`<br />`5.853 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                                                                       |    `10.68 kB`<br />`4.215 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `23.554 kB`<br />`7.644 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                                                                       |    `9.238 kB`<br />`3.815 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                                                                         |   `12.231 kB`<br />`4.925 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `115.941 kB`<br />`35.393 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             |  `119.14 kB`<br />`35.886 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                                                                               |   `10.898 kB`<br />`4.174 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 | `103.061 kB`<br />`31.573 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |   `10.576 kB`<br />`3.875 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                                                                    |     `19.7 kB`<br />`7.404 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                                                                  |   `15.655 kB`<br />`5.835 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |      `36.238 kB`<br />`12 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `14.361 kB`<br />`5.728 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `20.746 kB`<br />`7.299 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |   `32.07 kB`<br />`10.033 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |  `43.899 kB`<br />`12.362 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                                                                 |   `19.848 kB`<br />`6.384 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |  `32.562 kB`<br />`10.253 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |    `23.674 kB`<br />`7.83 kB` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/SingleThemeToken.fixture.js'>Single theme token import</abbr>                                                        |            `69 B`<br />`89 B` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `41.504 kB`<br />`14.622 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/8f420a1c55f283156b758bda1e7ef0591ec57a78'>8f420a1c55f283156b758bda1e7ef0591ec57a78</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/Avatar-mount_base.html">1659</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/Avatar-mount.html">1672</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/Button-mount_base.html">1184</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/Button-mount.html">1208</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/FluentProvider-mount_base.html">2024</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/FluentProvider-mount.html">2052</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/FluentProviderWithTheme-mount_base.html">722</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/FluentProviderWithTheme-mount.html">789</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">719</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">680</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">739</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">748</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/MakeStyles-mount_base.html">2518</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/MakeStyles-mount.html">2578</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/SpinButton-mount_base.html">3406</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24378/perf-test-react-components/SpinButton-mount.html">3383</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-theme-bdcdae2f-8e92-4d18-b7da-d8336334caa6.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Add fontWeightBold, add Body2, fix font weight for *Stronger typography",
```
Comment (ling1726) [inline change/@fluentui-react-theme-bdcdae2f-8e92-4d18-b7da-d8336334caa6.json:3]:
```suggestion
  "comment": "feat: Add fontWeightBold, add Body2, fix font weight for *Stronger typography",
```

File: change/@fluentui-react-text-01530fa1-545f-4a94-a9fb-00fa2be3dc3b.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "Add Body2",
```
Comment (ling1726) [inline change/@fluentui-react-text-01530fa1-545f-4a94-a9fb-00fa2be3dc3b.json:3]:
```suggestion
  "comment": "feat: Add Body2",
```

File: change/@fluentui-react-theme-sass-7ddd5d89-4d81-49e6-b347-a1f779b7dbf0.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Add fontWeightBold",
```
Comment (ling1726) [inline change/@fluentui-react-theme-sass-7ddd5d89-4d81-49e6-b347-a1f779b7dbf0.json:3]:
```suggestion
  "comment": "feat: Add fontWeightBold",
```

**Review by ling1726 — APPROVED** [review]

**Review by marcosmoura — APPROVED** [review]

**Pull request merged**